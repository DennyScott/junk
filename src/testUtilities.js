import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

export const rendersCorrectly =  component => {
  const wrapper = shallow(component);
  expect(wrapper.length).toBe(1);
}

export const matchesSnapshot = component => {
  const wrapper = shallow(component);
  expect(toJson(wrapper)).toMatchSnapshot();
}