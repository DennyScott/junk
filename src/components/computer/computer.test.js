import React from 'react';
import Computer from './computer';
import { rendersCorrectly, matchesSnapshot } from 'testUtilities';

const minProps = {};

const getMinComponent = () => <Computer {...minProps} />;

it('Computer renders correctly', () => {
  rendersCorrectly(getMinComponent(), expect);
});

it('Computer matches snapshot correctly', () => {
  matchesSnapshot(getMinComponent(), expect);
});
