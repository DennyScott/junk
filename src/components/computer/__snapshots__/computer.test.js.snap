// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Computer matches snapshot correctly 1`] = `
<styled.div>
  <Connect(Desktop) />
  <Connect(Taskbar) />
</styled.div>
`;
